src/components/learner-credit-management/BudgetAggregates.jsx

```diff
@@ -0,0 +1,76 @@
+import { useParams } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import {
+  Card,
+  Stack,
+  Badge,
+  Icon,
+} from '@edx/paragon';
+import { Info } from '@edx/paragon/icons';
+import { formatDate, getBudgetStatus } from './data';
+import useOffersBudgets from './data/hooks/useOffersBudgets';
+import LearnerCreditAggregateCards from './LearnerCreditAggregateCards';
+
+const BudgetAggregates = ({
+  enterpriseUUID,
+  offers,
+}) => {
+  const { budgetId } = useParams();
+  const {
+    isLoading: isLoadingBudgets,
+    allBudgets,
+  } = useOffersBudgets(enterpriseUUID, offers);
+  const budget = allBudgets?.find(item => item.id === budgetId);
+  const budgetLabel = getBudgetStatus(budget?.start, budget?.end);
+  const renderCardHeader = () => {
+    const subtitle = (
+      <Stack direction="horizontal" gap={2.5}>
+        <Badge variant={budgetLabel?.badgeVarient} className="m-0">{budgetLabel?.status}</Badge>
+        <span data-testid="offer-date">
+          {formatDate(budget?.start)} - {formatDate(budget?.end)}
+        </span>
+      </Stack>
+    );
+
+    return (
+      <Card.Header
+        title={budget?.displayName || 'Overview'}
+        subtitle={subtitle}
+      />
+    );
+  };
+  return (
+    <Card isLoading={isLoadingBudgets}>
+      {renderCardHeader()}
```

Comment:
NIT: Instead of using a function to render the card header, you can create a separate React component for the customized header

```
const CardHeader = () => {
  const subtitle = (
    <Stack direction="horizontal" gap={2.5}>
      <Badge variant={budgetLabel?.badgeVariant} className="m-0">
        {budgetLabel?.status}
      </Badge>
      <span data-testid="offer-date">
        {formatDate(budget?.start)} - {formatDate(budget?.end)}
      </span>
    </Stack>
  );

  return (
    <Card.Header title={budget?.displayName || 'Overview'} subtitle={subtitle} />
  );
};
```

and then use it like
`<CardHeader />`

Reply:
Actually, we are not supposed to use that component anywhere else. That's why I am using this as a function here. Also, we are following a similar pattern in other places like rendering values through functions.
 What do you think?